libs/designer/src/lib/core/BJSWorkflowProvider.tsx

```diff
@@ -14,15 +14,16 @@ export interface BJSWorkflowProviderProps {
   workflow: Workflow;
   runInstance?: LogicAppsV2.RunInstanceDefinition | null;
   children?: React.ReactNode;
+  shouldRecreateWorkflowGraph?: boolean;
 }
 
-const DataProviderInner: React.FC<BJSWorkflowProviderProps> = ({ workflow, children, runInstance }) => {
+const DataProviderInner: React.FC<BJSWorkflowProviderProps> = ({ workflow, children, runInstance, shouldRecreateWorkflowGraph }) => {
```

Comment:
Does this get called even if `shouldRecreateWorkflowGraph` is already `true`? That is, if the data provider is set up with `shouldRecreateWorkflowGraph=false`, then we set `shouldRecreateWorkflowGraph=true`, it will dispatch a render. What happens if we then need to re-render again?

Reply:
I have a feeling we should not use a prop here, and instead should have our own call to dispatch the initialization. @hartra344 ?